eva-accession-core/src/test/java/uk/ac/ebi/eva/accession/core/SubmittedVariantModelSummaryFunctionTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.eva.accession.core;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+public class SubmittedVariantModelSummaryFunctionTest {
+
+    private static final String ASSEMBLY_ACCESSION = "assembly";
+
+    private static final int TAXONOMY_ACCESSION = 1;
+
+    private static final String PROJECT_ACCESSION = "project";
+
+    private static final String CONTIG = "contig";
+
+    private static final int START = 1;
+
+    private static final String REF_A = "A";
+
+    private static final String ALT_T = "T";
+
+    private static final boolean SUPPORTED_BY_EVIDENCE = true;
+
+    private SubmittedVariantModelSummaryFunction summaryFunction;
+
+    @Before
+    public void setUp() {
+        summaryFunction = new SubmittedVariantModelSummaryFunction();
+    }
+
+    @Test
+    public void summaryFunctionMustBeIdempotent() {
```

Comment:
I don't think idempotent is the right word here. idempotence means `f(x) == f(f(x))`

Reply:
https://en.wikipedia.org/wiki/Idempotence#Computer_science_meaning

Reply:
alright then. not my favorite thing to do, though, taking concise words and use them with a different meaning. Damn computer scientist! they ruined computer science!